components/brave_ads/rust/string_strip_unittest.cc

```diff
@@ -1,17 +1,29 @@
-/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
  * This Source Code Form is subject to the terms of the Mozilla Public
  * License, v. 2.0. If a copy of the MPL was not distributed with this file,
  * You can obtain one at https://mozilla.org/MPL/2.0/. */
 
-#include "brave/components/brave_ads/core/internal/common/strings/string_strip_util.h"
-
+#include "brave/components/brave_ads/rust/lib.rs.h"
 #include "testing/gtest/include/gtest/gtest.h"
 
 // npm run test -- brave_unit_tests --filter=BraveAds*
 
 namespace brave_ads {
 
-TEST(BraveAdsStringStripUtilTest, StripNonAlphaCharactersFromEmptyContent) {
+namespace {
+
+std::string StripNonAlphaCharacters(const std::string& value) {
+  return static_cast<std::string>(brave_ads::strip_non_alpha_character(value));
+}
+
+std::string StripNonAlphaNumericCharacters(const std::string& value) {
+  return static_cast<std::string>(
+      brave_ads::strip_non_alpha_numeric_character(value));
+}
```

Comment:
Why `static_cast` here? The way I assumed it worked when we did `std::string(some_rust_string)` was that it use the copy construtor to duplicate the data while dropping the original so the `rust::String` dtor ran on the rust side of the ffi bridge. It seems like this risks deallocation on the C++ side instead.

I see we have this idiom is lots of places already, so I'm curious how it works?

Reply:
we are explicitly calling the conversion operator from `rust:String` here:
`explicit operator std::string() const;`

Reply:
I also flagged this somewhere else because doesn't CXXString directly convert to std::string? It seems silly to have these wrappers just to cast to std::string

Reply:
agreed, returning CXXString directly will implicitly calls this cast